File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-213927-34997/workdir/src/surveycrawler/ground.py
class Ground:
    """The type of ground the sonar detects at a position.

    It knows what happens to a crawler that tries to move to it or to turn while standing on it.
    """

    # moving

    def move_crawler_towards(self, a_crawler, a_direction):
        raise NotImplementedError()

    # facing

    def turn_crawler_clockwise(self, a_crawler):
        raise NotImplementedError()

    def turn_crawler_counter_clockwise(self, a_crawler):
        raise NotImplementedError()

File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260921-212555-19812/output/project/src/main/java/surveycrawler/Ground.java
package surveycrawler;

public abstract class Ground {

    // moving

    public abstract void moveCrawlerTowards(SurveyCrawler aCrawler, Point aDirection);

    // facing

    public abstract void turnCrawlerClockwise(SurveyCrawler aCrawler);

    public abstract void turnCrawlerCounterClockwise(SurveyCrawler aCrawler);
}
